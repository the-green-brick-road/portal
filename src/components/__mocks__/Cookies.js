/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }             from '@mui/material';

function Cookies(props) {

    /* --------- Gather inputs --------- */
    const { color } = props;

    /* -------- Defining theme --------- */
    return (
        <Box style={{ minHeight:'30px',zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> Cookies </Typography>
        </Box>
    );

}

export default Cookies;
