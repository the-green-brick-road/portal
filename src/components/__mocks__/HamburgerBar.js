/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerBar component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerBar(props) {

    /* --------- Gather inputs --------- */
    const { height, isNegative, theme={} }  = props
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }

    return (
        <Box style={{height:height, backgroundColor:color, width: '100%'}}>
            <Typography> HamburgerBar </Typography>
        </Box>
    );

}

export default HamburgerBar;
