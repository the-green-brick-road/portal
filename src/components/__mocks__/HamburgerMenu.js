/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerMenu component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerMenu(props) {

    /* --------- Gather inputs --------- */
    const { top , margin, isNegative, theme={} } = props;
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }
    const menuWidthString = `calc( 100vw - 2 * ${margin})`;

    return (
        <Box style={{ minHeight:'30px', width:menuWidthString, top:top, zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> HamburgerMenu </Typography>
        </Box>
    );

}

export default HamburgerMenu;