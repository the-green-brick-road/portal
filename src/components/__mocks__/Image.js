/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box }             from '@mui/material';

function Image(props) {

    /* --------- Gather inputs --------- */
    const { style={} }    = props;

    const local_style = JSON.parse(JSON.stringify(style))

    if (!('backgroundColor' in local_style)) { local_style['backgroundColor'] = 'blue' }

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Box style={local_style} />
    );
    /* eslint-enable padded-blocks */

}

export default Image;
