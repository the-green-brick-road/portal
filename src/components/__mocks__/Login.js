/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Login component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @29 may 2023
# Latest revision: 29 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box }             from '@mui/material';

function Login(props) {

    /* --------- Gather inputs --------- */
    const { style={} }    = props;

    const local_style = JSON.parse(JSON.stringify(style))

    if (!('backgroundColor' in local_style)) { local_style['backgroundColor'] = 'green' }

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Box style={local_style} />
    );
    /* eslint-enable padded-blocks */

}

export default Login;
