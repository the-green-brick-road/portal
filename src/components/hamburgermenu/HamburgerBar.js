/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Hamburger Bar component
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Local includes */
import HamburgerLogo             from './HamburgerLogo';
import HamburgerStack            from './HamburgerStack';
import HamburgerIcon             from './HamburgerIcon';

function HamburgerBar(props) {

    /* --------- Gather inputs --------- */
    const { height = '20px', isNegative = false, theme={palette:{primary:{main:'#000000'},secondary:{main:'#ffffff'}}} }  = props
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */

    let stackcolor = theme.palette.primary.main;
    if (isNegative) { stackcolor = theme.palette.common.white; }

    /* ----------- Define HTML --------- */
    return (
        <HamburgerStack placeholder="hamburgerbar" id="hamburgerbar" direction="row" alignItems="center" justifyContent="space-between" color={stackcolor} padding="0px" height={height}>
            <HamburgerLogo padding="20px" height={height} name='logo' />
            <HamburgerIcon width="30px" height="30px" color={stackcolor} />
        </HamburgerStack>
    );

}

export default HamburgerBar;
