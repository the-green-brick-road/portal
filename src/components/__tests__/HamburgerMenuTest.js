/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Analytics Provider test suite
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Test includes */
import { render, prettyDOM, act, screen, fireEvent } from '@testing-library/react'
import { expect, test}                               from '@jest/globals';

/* Material UI includes */
import { Science }                                   from '@mui/icons-material/';

/* Component under test */
import { default as HamburgerMenu }                  from '../../components/hamburgermenu/HamburgerMenu';

/* Mocks includes */
/* eslint-disable jest/no-mocks-import */
import { useMenu as mockUseMenu, MenuProvider as MockMenuProvider } from '../../providers/__mocks__/MenuProvider';
import { useLogging as mockUseLogging, LoggingProvider as MockLoggingProvider } from '../../providers/__mocks__/LoggingProvider';
import { default as MockRouter }  from '../../containers/__mocks__/Router';
import { default as MockLayout }  from '../../containers/__mocks__/Layout';
jest.mock('../../providers', () => ({
    useLogging: (() => { return mockUseLogging(); }),
    useMenu: (()    => { return mockUseMenu(); }),
}));
jest.mock("../../containers", () => ({
    Router: (props) => MockRouter(props),
    Layout: (props) => MockLayout(props),
}));
/* eslint-enable jest/no-mocks-import */

function MockLayoutWithMenu() {

    const theme = {
        "palette": {
            "common": {
                "black": "#444444",
                "white": "#555555",
            },
            "primary": {
                "main": "#666666",
            },
            "secondary" : {
                "main": "#777777",
            }
        }
    };

    return(
        <MockLayout>
            <HamburgerMenu height='20px' margin='30px' isNegative={false} itemHeight='10px' theme={theme}/>
        </MockLayout>
    )

};

describe("HamburgerMenu component" ,() => {

    const entries = [
        {
            "id":     "item1",
            "path":   "/",
            "icon":   Science,
            "target": "_top",
        },
        {
            "id": "item2",
            "icon" : Science,
            "subitems": [
                {
                    "id": "subitem21",
                    "path": "/test1",
                    "target": "_top",
                },
                {
                    "id": "subitem22",
                    "path": "/test2",
                    "target": "_top",
                }
            ],
        }
    ];

    const theme = {
        "palette": {
            "common": {
                "black": "#444444",
                "white": "#555555",
            },
            "primary": {
                "main": "#666666",
                "light": "#666667",
                "dark": "#666668",
            },
            "secondary" : {
                "main": "#777777",
                "light": "#777778",
                "dark": "#777779",
            },
            "disabled" : {
                "main": "#888888",
                "light": "#888889",
                "dark": "#888880",
            },
            "text" : {
                "primary": "#111111",
                "secondary": "#222222",
                "disabled": "#333333",
            },
        },
        "direction": "ltr",
        "typography": {
            "h1": {
                "fontSize": "2.5rem",
                "color": "#111111",
                "textTransform": "uppercase",
                "fontFamily": "Roboto Condensed",
            },
            "h2": {
                "fontSize": "2rem",
                "color": "#222222",
                "textTransform": "lowercase",
                "fontFamily": "Roboto Condensed",
            },
            "body1": {
                "fontSize": "14px",
                "color": "black",
                "fontFamily": "Roboto",
                "position": "relative",
                "textAlign": "justify",
            },
        },
    }

    test('Should display a hamburger menu', async () => {

        const view = render(
            <div>
                <MockLoggingProvider>
                    <MockMenuProvider entries={entries}>
                        <HamburgerMenu height='20px' margin='30px' isNegative={false} itemHeight='10px' theme={theme}/>
                    </MockMenuProvider>
                </MockLoggingProvider>
            </div>
        )
        const tree = prettyDOM(view.baseElement, Number.POSITIVE_INFINITY, {filterNode: () => true, escapeString: false,highlight: false});
        expect(tree).toMatchSnapshot();

    })

    test('Should display a negative hamburger menu', async () => {

        const view = render(
            <div>
                <MockLoggingProvider>
                    <MockMenuProvider entries={entries}>
                        <HamburgerMenu height='20px' margin='30px' isNegative={true} itemHeight='10px' theme={theme}/>
                    </MockMenuProvider>
                </MockLoggingProvider>
            </div>
        )
        const tree = prettyDOM(view.baseElement, Number.POSITIVE_INFINITY, {filterNode: () => true, escapeString: false,highlight: false});
        expect(tree).toMatchSnapshot();

    })


    test('Should display a hamburger menu even with no attributes', async () => {

        const view = render(
            <div>
                <MockLoggingProvider>
                    <MockMenuProvider entries={entries}>
                        <HamburgerMenu/>
                    </MockMenuProvider>
                </MockLoggingProvider>
            </div>
        )
        const tree = prettyDOM(view.baseElement, Number.POSITIVE_INFINITY, {filterNode: () => true, escapeString: false,highlight: false});
        expect(tree).toMatchSnapshot();

    })


    test('Should select menu item on click', async () => {


        const state = mockUseMenu();

        let view = null
        await act(async () => { // eslint-disable-line testing-library/no-unnecessary-act

            view = render(

                <div>
                    <MockLoggingProvider>
                        <MockMenuProvider entries={entries}>
                            <MockRouter layout={MockLayoutWithMenu} />
                        </MockMenuProvider>
                    </MockLoggingProvider>
                </div>

            )

        })


        const delay = ms => new Promise(res => setTimeout(res, ms));
        await delay(1000);

        await act(async () => {fireEvent.click(screen.getByText('item2'))}) // eslint-disable-line testing-library/no-unnecessary-act
        expect(state.selectEntry).toHaveBeenLastCalledWith('item2', true)
        await act(async () => {fireEvent.click(screen.getByText('item1'))}) // eslint-disable-line testing-library/no-unnecessary-act
        expect(state.selectEntry).toHaveBeenLastCalledWith('item1', false)


        expect(screen.getByText('item1').closest('a').getAttribute('href')).toBe('/')
        expect(screen.getByText('subitem21').closest('a').getAttribute('href')).toBe('/test1')
        expect(screen.getByText('subitem22').closest('a').getAttribute('href')).toBe('/test2')

    })


})
