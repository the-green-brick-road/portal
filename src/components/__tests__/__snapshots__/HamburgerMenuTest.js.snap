// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerMenu component Should display a hamburger menu 1`] = `
"<body>
  <div>
    <div>
      <div
        class=\\"MuiContainer-root MuiContainer-maxWidthLg css-25jtj5-MuiContainer-root\\"
        color=\\"#444444\\"
        left=\\"0\\"
        top=\\"95px\\"
        width=\\"calc( 100vw - 2 * 30px)\\"
      >
        <ul
          class=\\"MuiList-root MuiList-padding css-1jmgfrh-MuiList-root\\"
          padding=\\"0px\\"
          role=\\"menu\\"
          tabindex=\\"-1\\"
        >
          <div
            tabindex=\\"0\\"
          >
            <a
              class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-4m5fmp-MuiTypography-root-MuiLink-root\\"
              href=\\"/\\"
              name=\\"item1\\"
              target=\\"_top\\"
            >
              <li
                class=\\"MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-divider MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-divider css-6owcs9-MuiButtonBase-root-MuiMenuItem-root\\"
                color=\\"#666666\\"
                height=\\"10px\\"
                name=\\"item1\\"
                padding=\\"0px\\"
                role=\\"menuitem\\"
                tabindex=\\"-1\\"
              >
                <div
                  class=\\"MuiListItemIcon-root css-4i04uf-MuiListItemIcon-root\\"
                  color=\\"#666666\\"
                >
                  <svg
                    aria-hidden=\\"true\\"
                    class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                    data-testid=\\"ScienceIcon\\"
                    focusable=\\"false\\"
                    viewBox=\\"0 0 24 24\\"
                  >
                    <path
                      d=\\"M19.8 18.4 14 10.67V6.5l1.35-1.69c.26-.33.03-.81-.39-.81H9.04c-.42 0-.65.48-.39.81L10 6.5v4.17L4.2 18.4c-.49.66-.02 1.6.8 1.6h14c.82 0 1.29-.94.8-1.6z\\"
                    />
                  </svg>
                </div>
                <div
                  class=\\"MuiListItemText-root css-19f3duk-MuiListItemText-root\\"
                  color=\\"#666666\\"
                >
                  <span
                    class=\\"MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root\\"
                  >
                    item1
                  </span>
                </div>
                <span
                  class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
                />
              </li>
            </a>
          </div>
          <div>
            <li
              class=\\"MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-gtz6vn-MuiButtonBase-root-MuiMenuItem-root\\"
              color=\\"#666666\\"
              height=\\"10px\\"
              name=\\"item2\\"
              padding=\\"0px\\"
              role=\\"menuitem\\"
              tabindex=\\"-1\\"
            >
              <div
                class=\\"MuiListItemIcon-root css-4i04uf-MuiListItemIcon-root\\"
                color=\\"#666666\\"
              >
                <svg
                  aria-hidden=\\"true\\"
                  class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                  data-testid=\\"ScienceIcon\\"
                  focusable=\\"false\\"
                  viewBox=\\"0 0 24 24\\"
                >
                  <path
                    d=\\"M19.8 18.4 14 10.67V6.5l1.35-1.69c.26-.33.03-.81-.39-.81H9.04c-.42 0-.65.48-.39.81L10 6.5v4.17L4.2 18.4c-.49.66-.02 1.6.8 1.6h14c.82 0 1.29-.94.8-1.6z\\"
                  />
                </svg>
              </div>
              <div
                class=\\"MuiListItemText-root css-19f3duk-MuiListItemText-root\\"
                color=\\"#666666\\"
              >
                <span
                  class=\\"MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root\\"
                >
                  item2
                </span>
              </div>
              <span
                class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
              />
            </li>
            <a
              class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-4m5fmp-MuiTypography-root-MuiLink-root\\"
              href=\\"/test1\\"
              name=\\"subitem21\\"
              target=\\"_top\\"
            >
              <li
                class=\\"MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-1w5ofpo-MuiButtonBase-root-MuiMenuItem-root\\"
                color=\\"#666666\\"
                height=\\"10px\\"
                name=\\"item2\\"
                padding=\\"0px\\"
                role=\\"menuitem\\"
                tabindex=\\"-1\\"
              >
                <div
                  class=\\"MuiListItemText-root css-rjckym-MuiListItemText-root\\"
                  color=\\"#666666\\"
                >
                  <span
                    class=\\"MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root\\"
                  >
                    subitem21
                  </span>
                </div>
                <span
                  class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
                />
              </li>
            </a>
            <a
              class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-4m5fmp-MuiTypography-root-MuiLink-root\\"
              href=\\"/test2\\"
              name=\\"subitem22\\"
              target=\\"_top\\"
            >
              <li
                class=\\"MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-1w5ofpo-MuiButtonBase-root-MuiMenuItem-root\\"
                color=\\"#666666\\"
                height=\\"10px\\"
                name=\\"item2\\"
                padding=\\"0px\\"
                role=\\"menuitem\\"
                tabindex=\\"-1\\"
              >
                <div
                  class=\\"MuiListItemText-root css-rjckym-MuiListItemText-root\\"
                  color=\\"#666666\\"
                >
                  <span
                    class=\\"MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root\\"
                  >
                    subitem22
                  </span>
                </div>
                <span
                  class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
                />
              </li>
            </a>
            <hr
              class=\\"MuiDivider-root MuiDivider-fullWidth MuiDivider-light css-xetjfm-MuiDivider-root\\"
            />
          </div>
        </ul>
      </div>
    </div>
  </div>
</body>"
`;

exports[`HamburgerMenu component Should display a hamburger menu even with no attributes 1`] = `
"<body>
  <div>
    <div>
      <div
        class=\\"MuiContainer-root MuiContainer-maxWidthLg css-1w5hny9-MuiContainer-root\\"
        left=\\"0\\"
        top=\\"95px\\"
        width=\\"calc( 100vw - 2 * 20px)\\"
      >
        <ul
          class=\\"MuiList-root MuiList-padding css-1jmgfrh-MuiList-root\\"
          padding=\\"0px\\"
          role=\\"menu\\"
          tabindex=\\"-1\\"
        >
          <div
            tabindex=\\"0\\"
          >
            <a
              class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-4m5fmp-MuiTypography-root-MuiLink-root\\"
              href=\\"/\\"
              name=\\"item1\\"
              target=\\"_top\\"
            >
              <li
                class=\\"MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-divider MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-divider css-enhu5j-MuiButtonBase-root-MuiMenuItem-root\\"
                color=\\"#000000\\"
                height=\\"20px\\"
                name=\\"item1\\"
                padding=\\"0px\\"
                role=\\"menuitem\\"
                tabindex=\\"-1\\"
              >
                <div
                  class=\\"MuiListItemIcon-root css-uxqleb-MuiListItemIcon-root\\"
                  color=\\"#000000\\"
                >
                  <svg
                    aria-hidden=\\"true\\"
                    class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                    data-testid=\\"ScienceIcon\\"
                    focusable=\\"false\\"
                    viewBox=\\"0 0 24 24\\"
                  >
                    <path
                      d=\\"M19.8 18.4 14 10.67V6.5l1.35-1.69c.26-.33.03-.81-.39-.81H9.04c-.42 0-.65.48-.39.81L10 6.5v4.17L4.2 18.4c-.49.66-.02 1.6.8 1.6h14c.82 0 1.29-.94.8-1.6z\\"
                    />
                  </svg>
                </div>
                <div
                  class=\\"MuiListItemText-root css-16jtqit-MuiListItemText-root\\"
                  color=\\"#000000\\"
                >
                  <span
                    class=\\"MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root\\"
                  >
                    item1
                  </span>
                </div>
                <span
                  class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
                />
              </li>
            </a>
          </div>
          <div>
            <li
              class=\\"MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-2h6oef-MuiButtonBase-root-MuiMenuItem-root\\"
              color=\\"#000000\\"
              height=\\"20px\\"
              name=\\"item2\\"
              padding=\\"0px\\"
              role=\\"menuitem\\"
              tabindex=\\"-1\\"
            >
              <div
                class=\\"MuiListItemIcon-root css-uxqleb-MuiListItemIcon-root\\"
                color=\\"#000000\\"
              >
                <svg
                  aria-hidden=\\"true\\"
                  class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                  data-testid=\\"ScienceIcon\\"
                  focusable=\\"false\\"
                  viewBox=\\"0 0 24 24\\"
                >
                  <path
                    d=\\"M19.8 18.4 14 10.67V6.5l1.35-1.69c.26-.33.03-.81-.39-.81H9.04c-.42 0-.65.48-.39.81L10 6.5v4.17L4.2 18.4c-.49.66-.02 1.6.8 1.6h14c.82 0 1.29-.94.8-1.6z\\"
                  />
                </svg>
              </div>
              <div
                class=\\"MuiListItemText-root css-16jtqit-MuiListItemText-root\\"
                color=\\"#000000\\"
              >
                <span
                  class=\\"MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root\\"
                >
                  item2
                </span>
              </div>
              <span
                class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
              />
            </li>
            <a
              class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-4m5fmp-MuiTypography-root-MuiLink-root\\"
              href=\\"/test1\\"
              name=\\"subitem21\\"
              target=\\"_top\\"
            >
              <li
                class=\\"MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-1i2pbi7-MuiButtonBase-root-MuiMenuItem-root\\"
                color=\\"#000000\\"
                height=\\"20px\\"
                name=\\"item2\\"
                padding=\\"0px\\"
                role=\\"menuitem\\"
                tabindex=\\"-1\\"
              >
                <div
                  class=\\"MuiListItemText-root css-1jl7vcq-MuiListItemText-root\\"
                  color=\\"#000000\\"
                >
                  <span
                    class=\\"MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root\\"
                  >
                    subitem21
                  </span>
                </div>
                <span
                  class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
                />
              </li>
            </a>
            <a
              class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-4m5fmp-MuiTypography-root-MuiLink-root\\"
              href=\\"/test2\\"
              name=\\"subitem22\\"
              target=\\"_top\\"
            >
              <li
                class=\\"MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-1i2pbi7-MuiButtonBase-root-MuiMenuItem-root\\"
                color=\\"#000000\\"
                height=\\"20px\\"
                name=\\"item2\\"
                padding=\\"0px\\"
                role=\\"menuitem\\"
                tabindex=\\"-1\\"
              >
                <div
                  class=\\"MuiListItemText-root css-1jl7vcq-MuiListItemText-root\\"
                  color=\\"#000000\\"
                >
                  <span
                    class=\\"MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root\\"
                  >
                    subitem22
                  </span>
                </div>
                <span
                  class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
                />
              </li>
            </a>
            <hr
              class=\\"MuiDivider-root MuiDivider-fullWidth MuiDivider-light css-xetjfm-MuiDivider-root\\"
            />
          </div>
        </ul>
      </div>
    </div>
  </div>
</body>"
`;

exports[`HamburgerMenu component Should display a negative hamburger menu 1`] = `
"<body>
  <div>
    <div>
      <div
        class=\\"MuiContainer-root MuiContainer-maxWidthLg css-spm2fe-MuiContainer-root\\"
        color=\\"#666666\\"
        left=\\"0\\"
        top=\\"95px\\"
        width=\\"calc( 100vw - 2 * 30px)\\"
      >
        <ul
          class=\\"MuiList-root MuiList-padding css-1jmgfrh-MuiList-root\\"
          padding=\\"0px\\"
          role=\\"menu\\"
          tabindex=\\"-1\\"
        >
          <div
            tabindex=\\"0\\"
          >
            <a
              class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-4m5fmp-MuiTypography-root-MuiLink-root\\"
              href=\\"/\\"
              name=\\"item1\\"
              target=\\"_top\\"
            >
              <li
                class=\\"MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-divider MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-divider css-fugi7y-MuiButtonBase-root-MuiMenuItem-root\\"
                color=\\"#555555\\"
                height=\\"10px\\"
                name=\\"item1\\"
                padding=\\"0px\\"
                role=\\"menuitem\\"
                tabindex=\\"-1\\"
              >
                <div
                  class=\\"MuiListItemIcon-root css-1pm5neq-MuiListItemIcon-root\\"
                  color=\\"#555555\\"
                >
                  <svg
                    aria-hidden=\\"true\\"
                    class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                    data-testid=\\"ScienceIcon\\"
                    focusable=\\"false\\"
                    viewBox=\\"0 0 24 24\\"
                  >
                    <path
                      d=\\"M19.8 18.4 14 10.67V6.5l1.35-1.69c.26-.33.03-.81-.39-.81H9.04c-.42 0-.65.48-.39.81L10 6.5v4.17L4.2 18.4c-.49.66-.02 1.6.8 1.6h14c.82 0 1.29-.94.8-1.6z\\"
                    />
                  </svg>
                </div>
                <div
                  class=\\"MuiListItemText-root css-xcogc9-MuiListItemText-root\\"
                  color=\\"#555555\\"
                >
                  <span
                    class=\\"MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root\\"
                  >
                    item1
                  </span>
                </div>
                <span
                  class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
                />
              </li>
            </a>
          </div>
          <div>
            <li
              class=\\"MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-1tg625t-MuiButtonBase-root-MuiMenuItem-root\\"
              color=\\"#555555\\"
              height=\\"10px\\"
              name=\\"item2\\"
              padding=\\"0px\\"
              role=\\"menuitem\\"
              tabindex=\\"-1\\"
            >
              <div
                class=\\"MuiListItemIcon-root css-1pm5neq-MuiListItemIcon-root\\"
                color=\\"#555555\\"
              >
                <svg
                  aria-hidden=\\"true\\"
                  class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                  data-testid=\\"ScienceIcon\\"
                  focusable=\\"false\\"
                  viewBox=\\"0 0 24 24\\"
                >
                  <path
                    d=\\"M19.8 18.4 14 10.67V6.5l1.35-1.69c.26-.33.03-.81-.39-.81H9.04c-.42 0-.65.48-.39.81L10 6.5v4.17L4.2 18.4c-.49.66-.02 1.6.8 1.6h14c.82 0 1.29-.94.8-1.6z\\"
                  />
                </svg>
              </div>
              <div
                class=\\"MuiListItemText-root css-xcogc9-MuiListItemText-root\\"
                color=\\"#555555\\"
              >
                <span
                  class=\\"MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root\\"
                >
                  item2
                </span>
              </div>
              <span
                class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
              />
            </li>
            <a
              class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-4m5fmp-MuiTypography-root-MuiLink-root\\"
              href=\\"/test1\\"
              name=\\"subitem21\\"
              target=\\"_top\\"
            >
              <li
                class=\\"MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-88pf6u-MuiButtonBase-root-MuiMenuItem-root\\"
                color=\\"#555555\\"
                height=\\"10px\\"
                name=\\"item2\\"
                padding=\\"0px\\"
                role=\\"menuitem\\"
                tabindex=\\"-1\\"
              >
                <div
                  class=\\"MuiListItemText-root css-15kukmo-MuiListItemText-root\\"
                  color=\\"#555555\\"
                >
                  <span
                    class=\\"MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root\\"
                  >
                    subitem21
                  </span>
                </div>
                <span
                  class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
                />
              </li>
            </a>
            <a
              class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-4m5fmp-MuiTypography-root-MuiLink-root\\"
              href=\\"/test2\\"
              name=\\"subitem22\\"
              target=\\"_top\\"
            >
              <li
                class=\\"MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-88pf6u-MuiButtonBase-root-MuiMenuItem-root\\"
                color=\\"#555555\\"
                height=\\"10px\\"
                name=\\"item2\\"
                padding=\\"0px\\"
                role=\\"menuitem\\"
                tabindex=\\"-1\\"
              >
                <div
                  class=\\"MuiListItemText-root css-15kukmo-MuiListItemText-root\\"
                  color=\\"#555555\\"
                >
                  <span
                    class=\\"MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root\\"
                  >
                    subitem22
                  </span>
                </div>
                <span
                  class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
                />
              </li>
            </a>
            <hr
              class=\\"MuiDivider-root MuiDivider-fullWidth MuiDivider-light css-xetjfm-MuiDivider-root\\"
            />
          </div>
        </ul>
      </div>
    </div>
  </div>
</body>"
`;
